plugins/shodan/api_key.go

```diff
@@ -0,0 +1,57 @@
+package shodan
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://developer.shodan.io/api/requirements"),
+		ManagementURL: sdk.URL("https://account.shodan.io"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Shodan.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 32,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.TempFile(
+			provision.FieldAsFile(fieldname.APIKey),
+			provision.AtFixedPath("~/.config/shodan/api_key"),
+		),
+		Importer: importer.TryAll(
+			TryShodanConfigFile(),
+		)}
+}
+
+func TryShodanConfigFile() sdk.Importer {
+	return importer.TryFile("~/.config/shodan/api_key", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
```

Comment:
For many tools, the equivalent of ~/.config from Linux on MacOS is ~/Library/Application Support.

Is this the case here too?

Reply:
After checking the source code of the CLI tool (https://github.com/achillean/shodan-python/blob/master/shodan/cli/settings.py), it seems that the OS does not matter and the configuration file is stored in ~/.shodan/config if the ~/.shodan/ directory exists. Otherwise, it is stored in ~/.config/shodan/.
I modified the importers to try reading from both locations, and changed the provisioner to use the ~/.shodan/ directory.